### PR TITLE
fix-default to seurat method

### PR DESCRIPTION
**Pull request by eisascience**

### CODE REVIEW

File: R/Seurat_III.R [hunk 1]
```diff
@@ -1433,23 +1433,39 @@ FilterCloneNames <- function(seuratObj, minValue) {
...
-  ClustLS <- list()
-
-  for (lev in levels(AvlLevels)){
-    print(lev)
-    ClustLS[[lev]] <- as.data.frame(Matrix::rowMeans(GetAssayData(object = seuratObj, features = genes, slot = slot)[genes, colnames(seuratObj)[which(AvlLevels==lev)]  ]))
+  if(!varName %in% colnames(seuratObj@meta.data)) stop("varName provided is not in the meta.data of this object")
+  
+  if(SerMethod) {
+    Seurat::Idents(seuratObj) <- varName
+    if(slot=="scale.data") use.scale = T else use.scale = F
```
Comment (bbimber) [inline R/Seurat_III.R]: See the seurat code, https://github.com/satijalab/seurat/blob/b51801bc4b1a66aed5456473c9fe0be884994c93/R/utilities.R, we dont need to bother with use.scale and use.data.

File: R/Seurat_III.R [hunk 2]
```diff
@@ -1433,23 +1433,39 @@ FilterCloneNames <- function(seuratObj, minValue) {
...
+    if(slot=="counts") use.counts = T else use.counts = F
```
Comment (bbimber) [inline R/Seurat_III.R]: please be consistent w/ indentation/spacing.  for example, "if (", not "if("

File: R/Seurat_III.R [hunk 3]
```diff
@@ -1433,23 +1433,39 @@ FilterCloneNames <- function(seuratObj, minValue) {
...
+    ClustDF <- Seurat::AverageExpression(object=seuratObj, slot=slot, features = genes, use.scale = use.scale, use.counts = use.counts)[[1]]
+  } else {
+    
+    AvlLevels <- factor(as.character(FetchData(seuratObj, varName)[,1]))
```
Comment (bbimber) [inline R/Seurat_III.R:1449]: also whitespace after lines

File: R/Seurat_III.R [hunk 4]
```diff
@@ -1433,23 +1433,39 @@ FilterCloneNames <- function(seuratObj, minValue) {
...
+    
+    ClustLS <- list()
+    
+    for (lev in levels(AvlLevels)){
+      print(lev)
```
Comment (bbimber) [inline R/Seurat_III.R]: why is this printing?

File: R/Seurat_III.R [hunk 5]
```diff
@@ -1433,23 +1433,39 @@ FilterCloneNames <- function(seuratObj, minValue) {
 #' @param varName The resolution/ident to use
 #' @param genes A vector of genes to include
 #' @param slot The slot to use, passed to GetAssayData
+#' @param SerMethod This function now defaults to the base Seurat method set to T, unless when a simple average is needed set to F.
 #' @return A data.frame of avg expression per var
 #' @importFrom Matrix rowMeans
 #' @export
-AvgCellExprs <- function(seuratObj, varName = "ClusterNames_0.2", genes, slot = "scale.data"){
+AvgCellExprs <- function(seuratObj, varName = "ClusterNames_0.2", genes, slot = "data", SerMethod=T){
```
Comment (bbimber) [inline R/Seurat_III.R]: is there a specific reason to change our default slot?  i dont really care, but want to be sure this is deliberate.  Also, can you keep casing consistent, so serMethod, not SerMethod.  I might also call it "seuratMethod", but that's me.

**Review by bbimber — COMMENTED** [review]
@eisascience see comments

File: R/Seurat_III.R (hunk 4 above)
Comment (eisascience) [reply R/Seurat_III.R]: sanity check on the levels it is calculating... the default Seurat method has the verbose=T so does the same..

File: R/Seurat_III.R (hunk 5 above)
Comment (eisascience) [reply R/Seurat_III.R]: the default for Seurat is counts not scaled counts.. being consistent with their method..

**Pull request merged**